File: backend/e-commerce/src/main/java/fr/sncf/ecommerce/users/application/api/controllers/responses/UserResponseMapper.java
package fr.sncf.ecommerce.users.application.api.controllers.responses;

import org.springframework.stereotype.Component;
import fr.sncf.ecommerce.users.domain.models.User;

@Component
public class UserResponseMapper {

    public UserResponse map(User user) {
        return UserResponse.builder()
                .id(user.getId())
                .firstname(user.getFirstname())
                .lastname(user.getLastname())
                .email(user.getEmail())
                .role(user.getRole())
                .build();
    }
}

File: frontend/e-commerce/src/app/users/service/user-response.mapper.ts
import { User } from "../models/User";
import { UserRole } from "../models/UserRole";
import { UserResponse } from "./user-response";

export class UserResponseMapper {
map(userResponse: UserResponse): User{
    return new User({
        ...userResponse,
        role:userResponse.role as UserRole
    })
}
}